Steve Papacharalambous <stevep@freescale.com> 23 October 2006

Fix build powerpc64 build errors:
[snip]
g++ -O2  -DNDEBUG -Wall -W -Wshadow -Wpointer-arith -Wwrite-strings -pedantic -ffor-scope   -c getc_putc.cpp
getc_putc.cpp: In function `int main(int, char**)':
getc_putc.cpp:168: error: no matching function for call to `min(long unsigned
int, unsigned int)'
[/snip]


Patch downloaded from:
http://www.freebsd.org/cgi/cvsweb.cgi/ports/benchmarks/bonnie++/files/patch-getc_putc.cpp

Fix the build on 64-bit machines.
The STL provides a min(unsigned, unsigned) implimentation by default,
but not a min(unsigned long, unsigned) one.


powerpc64 architecture addition by Steve Papacharalambous.

diff --exclude CVS -uNr bonnie++-1.93c/getc_putc.cpp bonnie++-1.93c.modified/getc_putc.cpp
--- bonnie++-1.93c/getc_putc.cpp	Sun Dec  9 14:50:24 2001
+++ bonnie++-1.93c.modified/getc_putc.cpp	Mon Oct 23 11:06:50 2006
@@ -17,6 +17,12 @@
 #include "duration.h"
 #include "getc_putc.h"
 
+#if defined(__alpha__) || defined(__sparc64__) || defined(__amd64__) || defined(__ia64__) || defined(__powerpc64__)
+/* Work around for: line 168, no matching function for call to `min(long unsigned int, unsigned int)' */
+#include <sys/param.h>
+#define min MIN
+#endif
+
 static void usage()
 {
   fprintf(stderr, "usage:\n"
